src/ORM/UnionList.php

```diff
@@ -0,0 +1,226 @@
+<?php
+
+namespace SilverStripe\ORM;
+
+use SilverStripe\Dev\Debug;
+use SilverStripe\View\ViewableData;
+use LogicException;
+
+/**
+ * A list object that wraps around other SS_List instances.
+ */
+class UnionList extends ViewableData implements SS_List
+{
+    /**
+     * An array of SS_List instances
+     *
+     * @var array
+     */
+    protected $lists = array();
+
+    /**
+     * Create a new UnionList.
+     *
+     * @param array $lists - Lists to hold
+     */
+    public function __construct(array $lists)
+    {
+        $this->lists = array_values($lists);
+        parent::__construct();
+    }
+
+    /**
+     * Return the total number of items in each SS_List.
+     *
+     * @return int
+     */
+    public function count()
+    {
+        $count = 0;
+        foreach ($this->lists as $list) {
+            $count += $list->count();
+        }
+        return $count;
+    }
+
+     /**
+     * Returns true if one of the SS_List's have an item.
+     *
+     * @return bool
+     */
+    public function exists()
+    {
+        return $this->count() > 0;
```

Comment:
Thinking now. We should just wait to see how modules/user-code uses this in the wild, then optimize for the majority case. If people use ArrayList with it a lot, a nice optimization is just checking the $lists for an ArrayList->count() > 0 to avoid a DB call.